### PR TITLE
feat: switch permit_dask to True by default

### PR DESCRIPTION
**Pull request by lgray**
This PR makes dask-based nanoevents the default.

Currently there are some unrelated blockers for from (dask-)awkward related to numpy reducers.

### CODE REVIEW

**Comment by lgray** [comment]
@nsmith- I've had to mangle some parts of the nanoevents schema system to get this all to fit right. I think it's worth a look from you to determine a way to make the non-user facing parts of the interface decent again.

**Comment by lgray** [comment]
@nsmith- can you give this a look and lodge your complaints? May not get fixed in this PR but we can plan it for the future.

File: src/coffea/nanoevents/factory.py [hunk 1]
```diff
@@ -252,7 +245,7 @@ def from_root(
         access_log=None,
         iteritems_options={},
         use_ak_forth=True,
-        permit_dask=False,
+        permit_dask=True,
```
Comment (nsmith-) [inline src/coffea/nanoevents/factory.py]: This is a big API change, but it is coming in a yet-unreleased major coffea version change, so it is ok. It also helps users to be exposed to dask. My only complaint is that the argument name is still a bit strange to me, rather I would think `delayed=True` or `eager_mode=False` would be better?

File: tests/test_analysis_tools.py [hunk 2]
```diff
@@ -10,14 +10,14 @@
 fname = "tests/samples/nano_dy.root"
 eagerevents = NanoEventsFactory.from_root(
     {os.path.abspath(fname): "Events"},
-    schemaclass=NanoAODSchema.v6,
+    schemaclass=NanoAODSchema,
```
Comment (nsmith-) [inline tests/test_analysis_tools.py:13]: Does the `.v6` specialization still work but is just not needed?

Comment (lgray) [reply tests/test_analysis_tools.py:13]: Right now the `.vX` specializations are a little broken in their original implementation. The functional sub-classing thing you were doing doesn't quite fit into the daskiness.

File: src/coffea/nanoevents/factory.py (hunk 1 above)
Comment (lgray) [reply src/coffea/nanoevents/factory.py]: I think `delayed=True` makes more sense. I'll change this.

**Pull request merged**